containers/SelectedContract.ts

```diff
@@ -0,0 +1,62 @@
+import { createContainer } from "unstated-next";
+import { useState, useEffect } from "react";
+import { useRouter } from "next/router";
+import { ethers } from "ethers";
+import ContractList, { ContractInfo } from "./ContractList";
+
+function useSelectedContract() {
+  const router = useRouter();
+  const { getByAddress, loading } = ContractList.useContainer();
+  const [state, setState] = useState<{
+    address: string | null;
+    contract: ContractInfo | null;
+    isValid: boolean;
+  }>({ address: null, contract: null, isValid: false });
+
+  // this seeds the address based on URL query param. Should probably be its own hook.
+  useEffect(() => {
+    // we need to wait until the contract list is loaded
+    if (loading) return;
+    const queryAddress = router.query.address;
+    const isNewAddress = queryAddress !== state.address;
+
+    if (queryAddress && isNewAddress && typeof queryAddress === "string") {
+      setAddress(queryAddress);
+    }
+  }, [router!.query!.address, loading]);
```

Comment:
Interesting -- never seen this syntax before.

Reply:
i think this is wrong actually. was going for[ optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)! dont know why this works though